common/p_enemy.cpp

```diff
@@ -1660,23 +1804,27 @@ void A_Mushroom(AActor* actor)
 
 	A_Explode(actor); // make normal explosion
 
-	for (int i = -n; i <= n; i += 8) // launch mushroom cloud
+	if (serverside)
```

Comment:
I prefer an "early exit" style where instead of encasing most of the function in a big block, just return early if `!serverside`.  Also, keep in mind that in single-player, serverside AND clientside are true.